litestar/_signature/models/attrs_signature_model.py

```diff
@@ -238,30 +241,59 @@ def __init__(self) -> None:
 _converter: Converter = Converter()
 
 
-def _extract_exceptions(e: Any) -> list[ErrorMessage]:
+def _get_message_from_traceback(exc: StructureException, connection: ASGIConnection) -> ErrorMessage:
+    """Gets a message from an attrs validation error.
+
+    The key will be a dot-separated string of the attribute path that failed
+    validation. The message will be the error message from the exception (or the
+    last exception in the exception group, when applicable).
+
+    Args:
+        exc: The exception to get the message from
+        connection: The connection instance
+
+    Returns:
+        An error message
+    """
+
+    formatted_exception = traceback.format_exception(type(exc), value=exc, tb=exc.__traceback__)
+
+    # handles nested structures (e.g. a model attribute references another
+    # model) by going through all exceptions in the exception group
+    keys: list[str] = []
+    for line in formatted_exception:
+        if (match := key_re.findall(line)) and (key := match[0][0].strip()):
+            keys.append(key)
+
+    exception = exc
```

Comment:
Why?

Reply:
Oh that's a bit silly looking at it now. I can just set this to `None` and maybe rename it to be more clear. Unless you were asking about the block as a whole?

Reply:
Do you mean why not use `exc` directly and overwrite that instead of creating a new variable?

Reply:
Indeed

Reply:
Good callout, this wasn't needed and I've removed it.